### PR TITLE
SOU-L1TSV+VR: preparation: fixed block_coordinate_descent.py

### PR DESCRIPTION
**Pull request by atsuki-kuwata**
in `src/jaxsot/core/block_coordinate_descent.py`

Functions `QP_obj_xk(...)` & `QP_obj_ak(...):` are the same forms, so I integrated them into one function `QP_objective(...)`

### CODE REVIEW

**Comment by HajimeKawahara** [comment]
Thanks. Can you add a unit test for BCD?

File: tests/bcd/bcd_test.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    maxiter=100
+
+    for i in tqdm.tqdm(range(0,maxiter)):
+        for k in range(0,Nk):
+            X[k,:]=opt_ref_vr(k,Y,W,A,X,lamX)
+            A[:,k]=opt_map_l2(k,Y,W,A,X,lamA)
+
+    Y = np.dot(W, np.dot(A,X))
+    refs=492.2236970510863
+    assert np.abs(np.sum(Y)-refs)<1.e-16
```
Comment (HajimeKawahara) [inline tests/bcd/bcd_test.py]: Consider to use `pytest.approx`

File: src/jaxsot/core/block_coordinate_descent.py [hunk 2]
```diff
@@ -2,12 +2,8 @@
 import jax.numpy as jnp
 import jaxopt
 
-def QP_obj_xk(params,W_x,K,bx,shapeA0):
-    obj = 0.5*jnp.dot(params, jnp.dot((W_x+K),params)) + jnp.dot(bx,params)
-    return obj/shapeA0
-
-def QP_obj_ak(params,W_a,T_a,b,shapeA0):
-    obj = 0.5*jnp.dot(params, jnp.dot((W_a+T_a),params)) + jnp.dot(b,params)
+def QP_objective(params,W,b,shapeA0):
```
Comment (HajimeKawahara) [inline src/jaxsot/core/block_coordinate_descent.py:5]: Can you add a comment with Args and Returns?

File: tests/bcd/bcd_test.py (hunk 1 above)
Comment (atsuki-kuwata) [reply tests/bcd/bcd_test.py]: OK. I will try.

File: src/jaxsot/core/block_coordinate_descent.py (hunk 2 above)
Comment (atsuki-kuwata) [reply src/jaxsot/core/block_coordinate_descent.py:5]: Thank you. I will add the comment.

**Comment by atsuki-kuwata** [comment]
I fixed the files. Please confirm.

File: src/jaxsot/core/block_coordinate_descent.py (hunk 2 above)
Comment (HajimeKawahara) [reply src/jaxsot/core/block_coordinate_descent.py:5]: better to clarify the equation?
"""objective function, Q = 1/2 params^T W params + b^T params, for Quadratic Programming (QP)"""

Comment (atsuki-kuwata) [reply src/jaxsot/core/block_coordinate_descent.py:5]: Exactly. I modified.

**Comment by HajimeKawahara** [comment]
I confirmed that the test worked well, but it took  ~ 30 second, which was a bit long as a very simple unit test. Can you reduce the computation time by changing the parameters? @atsuki-kuwata

**Comment by atsuki-kuwata** [comment]
Thank you. I will try changing the parameters.

**Comment by atsuki-kuwata** [comment]
I changed the parameters. The test took ~4 seconds. Is it still long?

**Review by HajimeKawahara — APPROVED** [review]
It's fine. Thank you! @2ndmk2 Can you also review the PR? After the approval by @2ndmk2, I will merge the PR.

**Review by 2ndmk2 — APPROVED** [review]
Sorry for the late review. I can run a code, and It took 4secs as well.

**Pull request merged**